.plugin/generator/src/Visitors/NonAbstractPublicTypesVisitor.cs

```diff
@@ -0,0 +1,33 @@
+using Microsoft.Generator.CSharp.ClientModel;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using System.Reflection;
+
+namespace OpenAILibraryPlugin.Visitors;
+
+/// <summary>
+/// A visitor that removes a generated 'abstract' modifier if a custom code definition exists for the type and it
+/// doesn't provide 'abstract'.
+/// </summary>
+public class NonAbstractPublicTypesVisitor : ScmLibraryVisitor
+{
+    protected override TypeProvider Visit(TypeProvider type)
+    {
+        if (type.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Public)
+            && type.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Abstract)
+            && type.CustomCodeView?.DeclarationModifiers.HasFlag(TypeSignatureModifiers.Abstract) == false)
+        {
+            // Keep types defined in custom code without 'abstract' non-abstract
+
+            // Post-update:
```

Comment:
What update does this code refer to?

Reply:
That `type.Update` line (commented out) doesn't work yet (but will with a future version of the dependency) I updated these comments with clearer "To do" statements describing what's going on. The reflection code's just a workaround.